src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao{
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction =null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save movie to DB" + movie, exception);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            return Optional.ofNullable(session.get(Movie.class, id));
+        } catch (Exception exception) {
+            throw new DataProcessingException("Can't find movie by id:" + id, exception);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
```

Comment:
Use try with finally, where you use transaction and try with resources with Read operations. 
Let's use try with resources here)